src/main/java/com/roommatematcher/enterprise/RoomMateController.java

```diff
@@ -1,16 +1,69 @@
 package com.roommatematcher.enterprise;
 
+import com.roommatematcher.enterprise.dto.Profile;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.*;
 
+/**
+ * The controller for Roommate Matcher REST endpoints and web UI
+ * <p>
+ *     This class handles the CRUD operations for Roommate Matcher users,
+ *     via HTTP actions
+ * </p>
+ * <p>
+ *     This class also serves HTML based web pages,
+ *     for UI interactions with user profiles
+ * </p>
+ * @author <<yournamehere>>
```

Comment:
Since we now have good version control, which identifies who made what changes, the @author tag in JavaDoc has been deprecated.